File: problemset-new/003/00375-guess-number-higher-or-lower-ii/freedomDR.cpp
#include <bits/stdc++.h>

using namespace std;
// Forward declaration of guess API.
// @param num, your guess
// @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
int guess(int num);

class Solution {
public:
    int guessNumber(int n) {
        int lo = 1, hi = n;
        while(lo < hi) {
            int mid = lo + (hi-lo)/2;
            if(guess(mid) == 0) return mid;
            else if(guess(mid) == 1) lo = mid+1;
            else hi = mid;
        }
        return lo;
    }
};
int main()
{
    return 0;
}

File: problemset-new/003/00367-valid-perfect-square/freedomDR.go
package main

import "fmt"

func isPerfectSquare(num int) bool {
	lo, hi := 1, num/2
	for lo < hi {
		mid := lo + (hi-lo)/2
		if mid*mid < num {
			lo = mid + 1
		} else {
			hi = mid
		}
	}
	if lo*lo == num {
		return true
	} else {
		return false
	}
}

func main() {
	fmt.Println("vim-go")
}
